CRM/Price/BAO/PriceSet.php

```diff
@@ -1690,4 +1691,41 @@ public static function getExtendsOptions(): array {
     return $result;
   }
 
+  /**
+   * Get priceset ID of selected entity ID
+   *
+   * @param int $entityID
+   * @param string $entityTable
+   *
+   * @return int Price set ID
+   */
+  public static function getPriceSetFromEntityID($entityID, $entityTable = 'civicrm_contribution') {
```

Comment:
This shouldn't be needed - you already have a BAO_Order object so getPriceSetID() should work on it...